src/pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj

```diff
@@ -80,6 +80,43 @@
                               PreferredPackages="$(Id);@(RuntimeDependency)" />
   </Target>
 
+  <UsingTask TaskName="GetVersionFromAssembly" AssemblyFile="$(LocalBuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.Local.dll"/>
+
+  <Target Name="GenerateHashVersionsFile" BeforeTargets="GetPackageFiles">
+
+    <PropertyGroup>
+      <_SystemRuntimePath Condition="$([System.String]::new('%(File.Identity)').ToLowerInvariant().Contains('system.runtime.dll'))">%(File.Identity)</_SystemRuntimePath>
+      <_SystemPrivateCoreLibPath Condition="$([System.String]::new('%(File.Identity)').ToLowerInvariant().Contains('system.private.corelib.dll'))">%(File.Identity)</_SystemPrivateCoreLibPath>
+    </PropertyGroup>
+
+    <GetVersionFromAssembly PathToAssembly="$(_SystemRuntimePath)">
+      <Output TaskParameter="AssemblyHash" PropertyName="CoreFxHash" />
+    </GetVersionFromAssembly>
+    <GetVersionFromAssembly PathToAssembly="$(_SystemPrivateCoreLibPath)">
+      <Output TaskParameter="AssemblyHash" PropertyName="CoreClrHash" />
+    </GetVersionFromAssembly>
+
+    <Message Importance="Low" Text="The hash of corefx is $(CoreFxHash) taken from $(_SystemRuntimePath)" />
+    <Message Importance="Low" Text="The hash of coreclr is $(CoreClrHash) taken from $(_SystemPrivateCoreLibPath)" />
+
+    <ItemGroup>
+      <_VersionsFileLines Include="CoreSetup $(LatestCommit)" />
+      <_VersionsFileLines Include="CoreFx $(CoreFxHash)" />
+      <_VersionsFileLines Include="dotnet/versions/corefx $(CoreFxCurrentRef)" />
+      <_VersionsFileLines Include="CoreClr $(CoreClrHash)" />
+      <_VersionsFileLines Include="dotnet/versions/coreclr $(CoreClrCurrentRef)" />
```

Comment:
Is this the best format for consumers of the new file? Is the consumer an msbuild script, or something else?